src/channel.js

```diff
@@ -0,0 +1,554 @@
+import { VLWholeSlideMicroscopyImage, getFrameMapping } from './metadata.js';
+import { RenderingEngine } from './renderingEngine.js';
+import {
+    arraysEqual,
+  } from './utils.js';
+import TileImage from 'ol/source/TileImage';
+import TileLayer from 'ol/layer/Tile';
+
+/** Extracts value of Pixel Spacing attribute from metadata.
+ *
+ * @param {object} metadata - Metadata of a DICOM VL Whole Slide Microscopy Image instance
+ * @returns {number[]} Spacing between pixel columns and rows in millimeter
+ * @private
+ */
+ function _getPixelSpacing(metadata) {
+    const functionalGroup = metadata.SharedFunctionalGroupsSequence[0];
+    const pixelMeasures = functionalGroup.PixelMeasuresSequence[0];
+    return pixelMeasures.PixelSpacing;
+  }
+
+/** BlendingInformation for DICOM VL Whole Slide Microscopy Image instances
+ * with Image Type VOLUME.
+ *
+ * @class
+ * @memberof channel
+ */  
+
+class BlendingInformation {
+  /*
+  * An interface class to set/get the visualization/presentation parameters from a channel object
+  * @param {string} opticalPathIdentifier, ID of the channel 
+  * @param {number[]} color
+  * @param {number} opacity
+  * @param {number[]} thresholdValues
+  * @param {boolean} visible
+  */
+  constructor(
+    opticalPathIdentifier, 
+    color,
+    opacity,
+    thresholdValues,
+    visible) {
+
+    this.opticalPathIdentifier = opticalPathIdentifier;
+    this.color = [...color];
+    this.opacity = opacity;
+    this.thresholdValues = [...thresholdValues];
+    this.visible = visible;
+    }
+}
+
+/** Channel for DICOM VL Whole Slide Microscopy Image instances
+ * with Image Type VOLUME.
+ *
+ * @class
+ * @memberof channel
+ */
+
+class Channel {
```

Comment:
We don't want to expose this class as part of the public API, since it exposes OpenLayers types. It is currently not included in the `dicom-microscopy-viewer.js` file, so it has not actually been exposed. However, I was wondering whether we should move it into the `viewer.js` file or rename it to `_Channel` to emphasize that the class it "private". What do you think?

Reply:
renamed to _Channel